lib/Twig/TokenParser/CacheParser.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Twig\TokenParser;
+
+use Countable;
+use LogicException;
+use Pimcore\Twig\Node\CacheNode;
+use Pimcore\ValueObject\Collection\ArrayOfStrings;
+use Twig\Error\SyntaxError;
+use Twig\Node\Expression\ArrayExpression;
+use Twig\Node\Node;
+use Twig\Token;
+use Twig\TokenParser\AbstractTokenParser;
+use Twig\TokenStream;
+use ValueError;
+use function count;
+use function is_int;
+use function is_null;
+use function sprintf;
+
+/**
+ * @internal
+ */
+class CacheParser extends AbstractTokenParser
+{
+    public function parse(Token $token): Node
+    {
+        $lineno = $token->getLine();
+
+        $stream = $this->parser->getStream();
+        $key = $stream->expect(Token::STRING_TYPE, null, 'Please specify a cache key')->getValue();
+
+        $expressionParser = $this->parser->getExpressionParser();
+
+        $ttl = null;
+        $tags = [];
+        $force = false;
+        while ($stream->test(Token::NAME_TYPE)) {
+            $k = $stream->getCurrent()->getValue();
+            $stream->next();
+            $args = $expressionParser->parseArguments();
+
+            switch ($k) {
+                case 'ttl':
+                    $this->validateModifierHasSingleArgument($args, 'ttl', $stream);
+
+                    $ttl = $args->getNode('0')->getAttribute('value');
```

Comment:
casually noticed that it could be slightly optimized to reduce hardcoded text (eg. `ttl` with `$k`) by moving 
`validateModifierHasSingleArgument()` before the `switch` (which could be potentially swapped with a `match`) and maybe rename it to `validateModifier` and add a check if `$modifierName` is `in_array` of the valid options (ttl/tags/force).

```suggestion
            $this->validateModifier($args, $k, $stream);
            switch ($k) {
                case 'ttl':
                    $ttl = $args->getNode('0')->getAttribute('value');
```

Reply:
Thanks, I improved it like suggested. A match is not easily applicable, let's leave the switch for now!